lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2508,6 +2516,116 @@ signTransaction tl preferredLatestEra keyLookup (rootKey, rootPwd) utxo =
             keyLookup
             inputResolver
 
+type MakeRewardAccountBuilder k =
+    (k 'RootK XPrv, Passphrase "encryption") -> (XPrv, Passphrase "encryption")
+
+-- | Produce witnesses and construct a transaction from a given selection.
+--
+-- Requires the encryption passphrase in order to decrypt the root private key.
+-- Note that this doesn't broadcast the transaction to the network. In order to
+-- do so, use 'submitTx'.
+--
+buildAndSignTransactionNoSelection
+    :: forall (n :: NetworkDiscriminant)
+     . (Typeable n, DelegationAddress n ShelleyKey 'CredFromKeyK)
+    => Tracer IO WalletLog
+    -> TimeInterpreter (Either PastHorizonException)
+    -> DBLayer IO (SeqState n ShelleyKey) ShelleyKey
+    -> NetworkLayer IO Block
+    -> TransactionLayer ShelleyKey 'CredFromKeyK SealedTx
+    -> WalletId
+    -> AnyRecentEra
+    -> Passphrase "user"
+    -> TransactionCtx
+    -> IO (Tx, TxMeta, SealedTx)
+buildAndSignTransactionNoSelection
+    tr ti db@DBLayer{..} netLayer txLayer
+    walletId era pwd txContext = WriteTx.withRecentEra era $
```

Comment:
In a recent PR you replaced `txCtx` with `transactionContext` as name for values of type `TransactionCtx`. Let's not introduce a third name style.